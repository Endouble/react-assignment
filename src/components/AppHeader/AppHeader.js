import React from 'react';

import { Header } from 'semantic-ui-react';

const headerStyle = {
  backgroundColor: '#000000',
  padding: '15px'
}

const AppHeader = () => {
  return <header style={headerStyle}>
    <Header as='h1' inverted>
      <Header.Content>
        Star Wars Characters
        <Header.Subheader>a list from a far far away galaxy</Header.Subheader>
      </Header.Content>
    </Header>
  </header>
};

export default AppHeader;