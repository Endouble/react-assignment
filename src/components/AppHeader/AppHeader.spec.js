import React from 'react';
import { shallow } from 'enzyme';

import AppHeader from './AppHeader';

describe('Header Test', () => {
  let shallowHeader;

  const getShallowHeader = () => {
    if (!shallowHeader) {
      shallowHeader = shallow(<AppHeader />)
    }

    return shallowHeader;
  };

  beforeEach(() => {
    shallowHeader = undefined;
  });

  test('Component should render without crashing', () => {
    expect(getShallowHeader()).toBeDefined();
  });

  test('Component should render correctly', () => {
    expect(getShallowHeader()).toMatchSnapshot();
  });
});