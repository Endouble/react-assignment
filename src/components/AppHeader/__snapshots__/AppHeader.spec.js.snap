// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Header
        as="h1"
        inverted={true}
>
        <HeaderContent>
                Star Wars Characters
                <HeaderSubheader>
                        a list from a far far away galaxy
                </HeaderSubheader>
        </HeaderContent>
</Header>,
      "style": Object {
        "backgroundColor": "#000000",
        "padding": "15px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "h1",
        "children": <HeaderContent>
          Star Wars Characters
          <HeaderSubheader>
                    a list from a far far away galaxy
          </HeaderSubheader>
</HeaderContent>,
        "inverted": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "Star Wars Characters",
            <HeaderSubheader>
              a list from a far far away galaxy
</HeaderSubheader>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Star Wars Characters",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "a list from a far far away galaxy",
            },
            "ref": null,
            "rendered": "a list from a far far away galaxy",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Header
          as="h1"
          inverted={true}
>
          <HeaderContent>
                    Star Wars Characters
                    <HeaderSubheader>
                              a list from a far far away galaxy
                    </HeaderSubheader>
          </HeaderContent>
</Header>,
        "style": Object {
          "backgroundColor": "#000000",
          "padding": "15px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": <HeaderContent>
            Star Wars Characters
            <HeaderSubheader>
                        a list from a far far away galaxy
            </HeaderSubheader>
</HeaderContent>,
          "inverted": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "Star Wars Characters",
              <HeaderSubheader>
                a list from a far far away galaxy
</HeaderSubheader>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Star Wars Characters",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "a list from a far far away galaxy",
              },
              "ref": null,
              "rendered": "a list from a far far away galaxy",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
